engine/src/main/java/org/terasology/config/flexible/Setting.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import com.google.common.collect.Lists;
+import org.terasology.assets.ResourceUrn;
+import org.terasology.config.flexible.validators.SettingValueValidator;
+import org.terasology.utilities.subscribables.GeneralSubscribable;
+
+import java.beans.PropertyChangeEvent;
+import java.beans.PropertyChangeListener;
+import java.util.List;
+
+public class Setting<T> implements GeneralSubscribable {
+    private final T defaultValue;
+
+    private ResourceUrn id;
+
+    private String idString;
+    private T value;
+
+    private String name;
+
+    private String description;
+    private SettingValueValidator<T> valueValidator;
+    private List<PropertyChangeListener> subscribers;
+
+    public Setting(ResourceUrn id, T defaultValue, SettingValueValidator<T> valueValidator) {
```

Comment:
I think a default value should compulsorily be added to the `Setting` as all types contained in the `Setting` will have `null` as the default since they are classes.